testing/unit/test_runner.py

```diff
@@ -79,42 +77,45 @@ def test_reap_stale_ports(self):
         self.runner.target_set_error.assert_called()
 
 
-class TestRunnerSendingTestingResult(TestRunnerBase):
-    """Test case to test the runner sending device testing result to the device testing server"""
+class TestRunnerSendingQualificationResult(TestRunnerBase):
```

Comment:
This class name is too specific for a testing class.  This would be an individual test ("test_runner_sending_qualification_result") -- for the class name it should be a general bucket that encompasses a few things.  So, what's the next level up from this specific use case? What's common to all the helper methods and data that's in here?

Reply:
Yes, makes sense. The same test case can be reused when target devices complete the tests with different results (passed, error, etc.). What about TestTargetDeviceComplete?

Reply:
Oh, I see my confusion -- this class shouldn't be in this file at all. This class is specific to a kind of test so should be in an appropriate module file. This module (test_runner) is really only for the basics of running all kinds of tests, so shouldn't have any dependency on a specific (e.g. DeviceTestingServer).